Stripe/STPFormTextField.m

```diff
@@ -140,7 +140,7 @@ - (void)setAutoFormattingBehavior:(STPFormTextFieldAutoFormattingBehavior)autoFo
                 for (NSNumber *segmentLength in cardNumberFormat) {
                     NSUInteger segmentIndex = 0;
                     for (; index < attributedString.length && segmentIndex < [segmentLength unsignedIntegerValue]; index++, segmentIndex++) {
-                        if (segmentIndex + 1 == [segmentLength unsignedIntegerValue]) {
+                        if (index + 1 != attributedString.length && segmentIndex + 1 == [segmentLength unsignedIntegerValue]) {
```

Comment:
Ah, nice. This fixed the failing tests for the `intrinsicContentSize`?

Reply:
yeah and the extra space at the end of the card number